### ISSUE TITLE
[BUG] Cross-device link not permitted when moving background to System Backgrounds

### ISSUE THREAD

**Issue by isaac-mcfadyen** [SillyTavern/SillyTavern#2104]
### Environment

🐋 Docker

### System

Docker 25.0.3

### Version

neo-server 1.12.0-preview, but applicable to release/staging too

### Desktop Information

- Docker
- Node version v20.12.2
- llama.cpp server via API

### Describe the problem

When promoting a background from a Chat Background to a System Background via the upwards arrow on it, the following error is produced in Docker:
```
Error: EXDEV: cross-device link not permitted, rename 'uploads/ed7202961a0d89ee93f9066736b6ba0e' -> '/data/<username>/backgrounds/<name>.png'
    at Object.renameSync (node:fs:1031:11)
    at /home/node/app/src/endpoints/backgrounds.js:66:12
    at Layer.handle [as handle_request] (/home/node/app/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/node/app/node_modules/express/lib/router/route.js:149:13)
    at urlencodedParser (/home/node/app/node_modules/body-parser/lib/types/urlencoded.js:100:7)
    at Layer.handle [as handle_request] (/home/node/app/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/node/app/node_modules/express/lib/router/route.js:149:13)
    at Route.dispatch (/home/node/app/node_modules/express/lib/router/route.js:119:3)
    at Layer.handle [as handle_request] (/home/node/app/node_modules/express/lib/router/layer.js:95:5)
    at /home/node/app/node_modules/express/lib/router/index.js:284:15 {
  errno: -18,
  code: 'EXDEV',
  syscall: 'rename',
  path: 'uploads/ed7202961a0d89ee93f9066736b6ba0e',
  dest: '/data/<username>/backgrounds/<name>.png'
```

I've traced this to the following `fs.renameSync` call, which is present in all branches with a slightly different path format:
https://github.com/SillyTavern/SillyTavern/blob/59bb04f1b3d188e2d2ed8a6004b2d1210b7b1207/src/endpoints/backgrounds.js#L66

The issue is that `fs.renameSync` fails when renaming across volumes, as is the case with Docker because `uploads/` is not mounted on the same volume as `/data`, because it tries and fails to do a hard-link move (which is not allowed to cross device boundaries).

The solution is to do a `fs.copyFileSync()` followed by an `fs.unlinkSync()`, potentially only when the `fs.renameSync()` fails. 

I'm happy to create a PR but I noticed that `fs.renameSync()` is used in a lot of other places in the codebase, so wondered about pulling out the logic into a `renameOrCopy` function that first tries a rename, and then a copy/delete if it fails.

### Please tick the boxes

- [X] I have explained the issue clearly, and I included all relevant info
- [X] I have checked that this [issue hasn't already been raised](https://github.com/SillyTavern/SillyTavern/issues?q=is%3Aissue)
- [X] I have checked the [docs](https://docs.sillytavern.app/) ![important](https://img.shields.io/badge/Important!-F6094E)

**Comment by Cohee1207** [comment]
> The solution is to do a fs.copyFileSync() followed by an fs.unlinkSync()

I vote for that, no need to build fallbacks if this can just be a universal and reliable operation.

**Comment by isaac-mcfadyen** [comment]
I'm happy to PR then, if that works?

**Comment by Cohee1207** [comment]
Sure, go ahead.

**Comment by isaac-mcfadyen** [comment]
Closed by #2105
